sdk/identity/identity/samples/AzureCacheForRedis/ioredis.md

```diff
@@ -0,0 +1,273 @@
+
+##### Migration Guidance
+
+When migrating your existing your application code to authenticate with Azure AD, you need to replace the password input with the Azure AD token.
+Integrate the logic in your application code to fetch an Azure AD access token via the Azure Identity library. Store the token in a token cache, as shown below. Replace the token with the password configuring/retrieving logic in your application code.
+
+```ts
+import Redis from "ioredis";
+import { DefaultAzureCredential, TokenCredential } from "@azure/identity";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+async function returnPassword(credential: TokenCredential) {
+  try {
+    // The scope will be changed for Azure AD Public Preview
+    const redisScope = "https://*.cacheinfra.windows.net:10225/appid/.default"
+
+    // Fetch an Azure AD token to be used for authentication. This token will be used as the password.
+    let accessTokenObject = await credential.getToken(redisScope);
+    return accessTokenObject;
+  } catch (e) {
+    throw e;
+  }
+}
+
+async function main() {
+  // Construct a Token Credential from Identity library
+  const credential = new DefaultAzureCredential();
+  let accessTokenObject = await returnPassword(credential);
+  // Create ioredis client and connect to the Azure Cache for Redis over the TLS port using the access token as password.
+  let redis = new Redis({
+    username: process.env.REDIS_SERVICE_PRINCIPAL_NAME,
+    password: accessTokenObject.token,
+    tls: {
+      host: process.env.REDIS_HOSTNAME,
+      port: 6380,
+    },
+    keepAlive: 0
+  });
+
+  const id = setInterval(async () => {
+    // Check for password expiry
+    console.log("checking for expiration,");
+    if ((accessTokenObject.expiresOnTimestamp- 120*1000) <= Date.now()) {
+      accessTokenObject = await returnPassword(credential);
+    }
+  }, 1000);
+
+  for (let i = 0; i < 3; i++) {
+    try {
+      // Set a value against your key in the Azure Redis Cache.
+      await redis.set("Az:mykey", "value123");
+      // Fetch value of your key in the Azure Redis Cache.
+      console.log("redis key:", await redis.get("Az:mykey"));
+      break;
+    } catch (e) {
+      console.log("error during redis get", e.toString());
+      if ((accessTokenObject.expiresOnTimestamp <= Date.now())|| (redis.status === "end" || "close") ) {
+        redis.disconnect();
+        redis = new Redis({
+          username: process.env.REDIS_SERVICE_PRINCIPAL_NAME,
+          password: accessTokenObject.token,
+          tls: {
+            host: process.env.REDIS_HOSTNAME,
+            port: 6380,
+          },
+          keepAlive: 0
+        });
+      }
+    }
+    await sleep(1000);
```

Comment:
What's the purpose of the `sleep` here? The previous sample doesn't do this.

Reply:
removed it